feat: move credential docs into concept section (#85)
KILTprotocol/docs: docs/concepts/02_Credentials/04_attestation.md

---
id: attestation
title: Attestation
---

To create an attestation, the attester first receives a message with a `RequestForAttestation` object from the claimer.

The attester then decrypts the message and checks the validity of the content, ensuring that:

- the structure of the `RequestForAttestation` object is correct
- the owner of the `RequestForAttestation` is the sender of the message

To create the attestation, the attester:

- manually verifies the data inside the claim by checking that all the data fields match the requirements of the attestation
- creates an attestation object from the root hash of the RequestForAttestation and the attester’s address.
- stores the attestation hash on the chain by submitting a new transaction with the attester’s signature.

Once this is done, the attester:

> - creates an `AttestedClaim` object from the `RequestForAttestation` and the attestation objects

Reviewer: ```suggestion
- creates a `Credential` object from the `RequestForAttestation` and the attestation objects
```

> - encrypts the `AttestedClaim` object

Reviewer: ```suggestion
- encrypts the `Credential` object
```

Reviewer: In the same way we have recommended to use encryption for the claimer, we should do the same here. I.e., we should not include encryption as a step in the process, but should say to send the object back to the claimer (even thought this process will soon change). We could then mention that it is recommended to use the previously established encrypted channel, if one was created.

> - sends the encrypted `AttestedClaim` object back to the claimer.

Reviewer: ```suggestion
- sends the encrypted `Credential` object back to the claimer.
```
